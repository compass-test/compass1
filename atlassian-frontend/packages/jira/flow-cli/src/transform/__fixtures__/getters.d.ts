export default class TestGetters {
  constructor();
  get dom(): HTMLElement | undefined;
  get performanceOptions(): {
    enabled: boolean;
    samplingRate: number;
    slowThreshold: number;
  };
}
