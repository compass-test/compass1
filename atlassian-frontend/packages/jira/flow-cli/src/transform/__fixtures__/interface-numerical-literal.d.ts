export interface ListItemArray
  extends Array<
    Paragraph | OrderedList | BulletList | MediaSingle | CodeBlock
  > {
  0: Paragraph | MediaSingle | CodeBlock;
}
