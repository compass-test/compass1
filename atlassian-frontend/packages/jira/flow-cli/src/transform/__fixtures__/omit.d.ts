export interface A {
  foo: string;
  bar: number;
  baz: string;
}

export interface B {
  bfoo: string;
  bbar: number;
  bbaz: number;
}

export interface C extends A, B {
  cfoo: string;
}

export interface D extends Array<A>, B {
  cfoo: string;
}

export interface OptionalAndComplexSignatures {
  status: CardStatus;
  isCardVisible: boolean;
  previewOrientation: number;
  isPlayingFile: boolean;
  mediaViewerSelectedItem?: Identifier;
  metadata?: FileDetails;
  dataURI?: string;
  progress?: number;
  error?: Error;
}

export type OneKeyOmit = Omit<OptionalAndComplexSignatures, 'error'>;
export type OmitWithKeyOfOperator = Omit<OptionalAndComplexSignatures, keyof C>;
export type OmitWithTypeReference = Omit<
  OptionalAndComplexSignatures,
  Pick<A & B, 'bar' | 'bfoo'>
>;
export type MultipleKeyOmit = Omit<
  OptionalAndComplexSignatures,
  'error' | 'isCardVisible'
>;

export type OmitWithInheritInterface = Omit<C, 'cfoo'>;
export type OmitWithInheritInterfaceAndComplexTypeInHeritage = Omit<D, 'cfoo'>;
