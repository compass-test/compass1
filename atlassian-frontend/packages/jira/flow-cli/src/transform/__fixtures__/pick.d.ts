export interface A {
  foo: string;
  bar: number;
  baz: string;
}

export interface B {
  bfoo: string;
  bbar: number;
  bbaz: number;
}

export interface D extends A, B {
  dfoo: string;
}

export interface OptionalAndComplexSignatures {
  status: CardStatus;
  isCardVisible: boolean;
  previewOrientation: number;
  isPlayingFile: boolean;
  mediaViewerSelectedItem?: Identifier;
  metadata?: FileDetails;
  dataURI?: string;
  progress?: number;
  error?: Error;
}

export type SimplePick = Pick<A, 'foo'>;
export type NotFoundPick = Pick<A, 'Blub'>;
export type NestedPick = Pick<Pick<B, 'bbar' | 'bfoo'>, 'bbar'>;
export type PickInDefinition = Pick<C, 'cfoo'>;

export interface C {
  cfoo: Pick<B, 'bfoo' | 'bbar'>;
  cbar: number;
  cbaz: number;
}

export type IntersectionPick = Pick<A & B, 'bar' | 'bfoo'>;
export type OptionalAndComplexPropertyPick = Pick<
  OptionalAndComplexSignatures,
  'status' | 'metadata'
>;

export type PickIncludedOtherUtilities = Pick<
  OptionalAndComplexSignatures,
  Exclude<keyof OptionalAndComplexSignatures, 'isCardVisible'>
>;

export type PickWithDefinitionsInHeritageClaue = Pick<
  D,
  'dfoo' | 'bfoo' | 'foo'
>;

export interface RadioProps extends WithAnalyticsEventsProps {
  /** the aria-label attribute associated with the radio element */
  ariaLabel?: string;
  /** Field disabled */
  isDisabled?: boolean;
  /** Marks this as a required field */
  isRequired?: boolean;
  /** Field is invalid */
  isInvalid?: boolean;
  /** Set the field as checked */
  isChecked?: boolean;
  /** The label value for the input rendered to the dom */
  label?: ReactNode;
  /** Field name */
  name?: string;
  /** onChange event handler, passed into the props of each Radio Component instantiated within RadioGroup */
  onChange: React.ChangeEventHandler<HTMLInputElement>;
  onBlur?: React.FocusEventHandler<HTMLInputElement>;
  onFocus?: React.FocusEventHandler<HTMLInputElement>;
  onMouseDown?: React.MouseEventHandler;
  onMouseUp?: React.MouseEventHandler;
  onMouseEnter?: React.MouseEventHandler;
  onMouseLeave?: React.MouseEventHandler;
  /** onInvalid event handler, passed into the props of each Radio Component instantiated within RadioGroup */
  onInvalid?: (e: SyntheticEvent<any>) => void;
  /** Field value */
  value?: RadioValue;
  /**
      A `testId` prop is provided for specified elements, which is a unique string that appears as a data attribute `data-testid` in the rendered code, serving as a hook for automated tests
      we have 2 different testid generated based on the one you pass to the Radio component:
      - `{testId}--hidden-radio` to check if it got changed to checked/unchecked.
      - `{testId}--radio-label` to click the input, because in IE11 the input has opacity: 0 and can't be interacted.
    */
  testId?: string;
}
