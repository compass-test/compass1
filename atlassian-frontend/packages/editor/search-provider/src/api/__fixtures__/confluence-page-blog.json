{
  "scopes": [
    {
      "id": "confluence.page,blogpost",
      "results": [
        {
          "content": {
            "id": "1538426521",
            "type": "page"
          },
          "container": {
            "title": "Editor Media",
            "displayUrl": "/spaces/EM"
          },
          "title": "Editor Media Planning - 2020-07-07",
          "url": "/spaces/EM/pages/1538426521/Editor+Media+Planning+-+2020-07-07",
          "baseUrl": "https://product-fabric.atlassian.net/wiki",
          "iconCssClass": "aui-icon aui-icon-small aui-iconfont-page-default",
          "entityType": "content",
          "lastModified": "2020-07-07T03:24:39+00:00",
          "excerpt": ""
        },
        {
          "content": {
            "id": "873989347",
            "type": "page"
          },
          "container": {
            "title": "Media Platform Team",
            "displayUrl": "/spaces/FIL"
          },
          "title": "Weekly Media Triad Meeting",
          "url": "/spaces/FIL/pages/873989347/Weekly+Media+Triad+Meeting",
          "baseUrl": "https://product-fabric.atlassian.net/wiki",
          "iconCssClass": "aui-icon aui-icon-small aui-iconfont-page-default",
          "entityType": "content",
          "lastModified": "2020-07-05T23:30:42+00:00",
          "excerpt": ""
        },
        {
          "content": {
            "id": "1356074858",
            "type": "page"
          },
          "container": {
            "title": "Editor Media",
            "displayUrl": "/spaces/EM"
          },
          "title": "Media test Media single",
          "url": "/spaces/EM/pages/1356074858/Media+test+Media+single",
          "baseUrl": "https://product-fabric.atlassian.net/wiki",
          "iconCssClass": "aui-icon aui-icon-small aui-iconfont-page-default",
          "entityType": "content",
          "lastModified": "2020-05-05T06:06:28+00:00",
          "excerpt": ""
        },
        {
          "content": {
            "id": "1385631549",
            "type": "page"
          },
          "container": {
            "title": "Editor Media",
            "displayUrl": "/spaces/EM"
          },
          "title": "Editor Media OKRs",
          "url": "/spaces/EM/pages/1385631549/Editor+Media+OKRs",
          "baseUrl": "https://product-fabric.atlassian.net/wiki",
          "iconCssClass": "aui-icon aui-icon-small aui-iconfont-page-default",
          "entityType": "content",
          "lastModified": "2020-06-09T02:21:27+00:00",
          "excerpt": ""
        },
        {
          "content": {
            "id": "998082697",
            "type": "superpage"
          },
          "container": {
            "title": "Editor",
            "displayUrl": "/spaces/E"
          },
          "title": "Media linking",
          "url": "/spaces/E/pages/998082697/Media+linking",
          "baseUrl": "https://product-fabric.atlassian.net/wiki",
          "iconCssClass": "aui-icon aui-icon-small aui-iconfont-page-default",
          "entityType": "content",
          "lastModified": "2020-04-15T03:00:05+00:00",
          "excerpt": ""
        },
        {
          "content": {
            "id": "998082697",
            "type": "blogpost"
          },
          "container": {
            "title": "Editor",
            "displayUrl": "/spaces/E"
          },
          "title": "Media linking",
          "url": "/spaces/E/pages/998082697/Media+linking",
          "baseUrl": "https://product-fabric.atlassian.net/wiki",
          "iconCssClass": "aui-icon aui-icon-small aui-iconfont-page-default",
          "entityType": "content",
          "lastModified": "2020-04-15T03:00:05+00:00",
          "excerpt": ""
        }
      ],
      "size": 4871,
      "abTest": {
        "abTestId": "default",
        "controlId": "default",
        "experimentId": "content-searcher"
      }
    }
  ]
}
