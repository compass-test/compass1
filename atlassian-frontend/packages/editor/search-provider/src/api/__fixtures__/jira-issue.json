{
  "scopes": [
    {
      "id": "jira.issue",
      "results": [
        {
          "id": "128578",
          "name": "FE error causing publishing errors - External Media Node must have a 'url' attribute",
          "url": "https://product-fabric.atlassian.net/browse/EDM-627",
          "attributes": {
            "@type": "issue",
            "key": "EDM-627",
            "issueTypeId": "10002",
            "issueTypeName": "Task",
            "containerId": "13055",
            "container": {
              "title": "Editor Media",
              "id": "13055"
            },
            "updated": "2020-07-08T14:59:56.112+1000",
            "avatar": {
              "url": "https://product-fabric.atlassian.net/secure/viewavatar?size=medium&avatarId=10318&avatarType=issuetype"
            }
          }
        },
        {
          "id": "162716",
          "name": "[BE] [SLA] [Edit Page] External media nodes are missing 'url' attribute",
          "url": "https://product-fabric.atlassian.net/browse/CERN-374",
          "attributes": {
            "@type": "issue",
            "key": "CERN-374",
            "issueTypeId": "11415",
            "issueTypeName": "Bug",
            "containerId": "12981",
            "container": {
              "title": "Confluence Editor - Reactor Core",
              "id": "12981"
            },
            "updated": "2020-07-08T00:43:47.608+1000",
            "avatar": {
              "url": "https://product-fabric.atlassian.net/secure/viewavatar?size=medium&avatarId=10303&avatarType=issuetype"
            }
          }
        },
        {
          "id": "167819",
          "name": "Media file fails to insert",
          "url": "https://product-fabric.atlassian.net/browse/EDM-833",
          "attributes": {
            "@type": "issue",
            "key": "EDM-833",
            "issueTypeId": "10002",
            "issueTypeName": "Story",
            "containerId": "13055",
            "container": {
              "title": "",
              "id": "13055"
            },
            "updated": "2020-07-08T09:02:18.334+1000",
            "avatar": {
              "url": "https://product-fabric.atlassian.net/secure/viewavatar?size=medium&avatarId=10318&avatarType=issuetype"
            }
          }
        },
        {
          "id": "169212",
          "name": "Create Editor Media amplitude dashboard",
          "url": "https://product-fabric.atlassian.net/browse/EDM-848",
          "attributes": {
            "@type": "issue",
            "key": "EDM-848",
            "issueTypeId": "10002",
            "issueTypeName": "Task",
            "containerId": "13055",
            "container": {
              "title": "Editor Media",
              "id": "13055"
            },
            "updated": "2020-07-08T09:02:18.548+1000",
            "avatar": {
              "url": "https://product-fabric.atlassian.net/secure/viewavatar?size=medium&avatarId=10318&avatarType=issuetype"
            }
          }
        },
        {
          "id": "170464",
          "name": "We've had a bunch of performance issues in Smart Links - all fixed, but we should monitor/instrument them (4 votes)",
          "url": "https://product-fabric.atlassian.net/browse/EDM-888",
          "attributes": {
            "@type": "issue",
            "key": "EDM-888",
            "issueTypeId": "10002",
            "issueTypeName": "",
            "containerId": "13055",
            "container": {
              "title": "Editor Media",
              "id": "13055"
            },
            "updated": "2020-06-19T16:00:12.589+1000",
            "avatar": {
              "url": "https://product-fabric.atlassian.net/secure/viewavatar?size=medium&avatarId=10318&avatarType=issuetype"
            }
          }
        }
      ],
      "size": 4654,
      "abTest": {
        "abTestId": "filters-experiment",
        "controlId": "filters-experiment",
        "experimentId": "filters-experiment"
      }
    }
  ]
}
