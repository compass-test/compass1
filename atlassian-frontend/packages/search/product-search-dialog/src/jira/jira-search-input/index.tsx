import * as JiraSearchInputImports from './jira-search-input';

export { JiraSearchInput } from './jira-search-input';

export type Props = JiraSearchInputImports.Props;
