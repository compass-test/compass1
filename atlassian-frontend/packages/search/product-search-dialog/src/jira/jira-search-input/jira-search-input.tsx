import React, { FunctionComponent } from 'react';
import { injectIntl, InjectedIntlProps } from 'react-intl';

import {
  ProductSearchInput,
  BaseInputProps,
} from '../../common/product-search-input';
import { messages } from '../../messages';
import { AdvancedSearchLinkSubjectId } from '../../common/analytics/events';
import { useFiltersAdvancedSearchUrlFactory } from '../jira-advanced-search-url-factory';

interface AdditionalProps {
  debounceTime: number;
  forwardRef: React.Ref<HTMLInputElement>;
  isLoading: boolean; // TODO: Move this into the analytic context
  onNavigate: (href: string, event: React.MouseEvent | KeyboardEvent) => void;
  onOpen: () => void;
}

export type Props = BaseInputProps & AdditionalProps;

const SearchInputBase: FunctionComponent<Props & InjectedIntlProps> = ({
  intl,
  ...rest
}) => {
  const advancedIssueSearchUrlFactory = useFiltersAdvancedSearchUrlFactory();
  const advancedSearchUrl = advancedIssueSearchUrlFactory(rest.value);

  return (
    <ProductSearchInput
      {...rest}
      advancedSearchURL={advancedSearchUrl}
      actionSubjectId={AdvancedSearchLinkSubjectId.JIRA_ISSUES}
      expandedPlaceholder={intl.formatMessage(messages.jira_search_placeholder)}
      collapsedPlaceholder={intl.formatMessage(
        messages.common_search_input_collapsed_placeholder,
      )}
    />
  );
};

// This is used for testing, do not consume this component in the UI
export const JiraSearchInput = injectIntl(SearchInputBase);
