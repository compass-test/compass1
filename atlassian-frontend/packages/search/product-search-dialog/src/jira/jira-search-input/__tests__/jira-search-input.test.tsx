import React from 'react';

import { mountWithIntl } from '../../../__tests__/__fixtures__/intl-test-helpers';

import { JiraSearchInput } from '../jira-search-input';

jest.mock('../../../common/analytics', () => {
  const fireAnalyticsEvent = jest.fn();

  return Object.assign({}, jest.requireActual('../../../common/analytics'), {
    useAnalytics: () => ({
      fireAnalyticsEvent,
    }),
  });
});

jest.mock('../../../common/product-search-input', () => {
  return { ProductSearchInput: (props: any) => <div {...props} /> };
});

const commonProps = {
  onOpen: () => {},
  debounceTime: 0,
  forwardRef: null,
  onNavigate: jest.fn(),
  isLoading: true,
  value: '',
};

describe('<SearchInput />', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('matches snapshot when expanded', () => {
    const wrapper = mountWithIntl(
      <JiraSearchInput {...commonProps} isExpanded />,
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('matches snapshot when collapsed', () => {
    const wrapper = mountWithIntl(
      <JiraSearchInput {...commonProps} isExpanded={false} />,
    );

    expect(wrapper).toMatchSnapshot();
  });
});
