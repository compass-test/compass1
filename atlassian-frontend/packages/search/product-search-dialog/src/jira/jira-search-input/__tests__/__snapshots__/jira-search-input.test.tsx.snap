// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchInput /> matches snapshot when collapsed 1`] = `
<SearchInputBase
  debounceTime={0}
  forwardRef={null}
  intl={
    Object {
      "defaultFormats": Object {},
      "defaultLocale": "en",
      "formatDate": [Function],
      "formatHTMLMessage": [Function],
      "formatMessage": [Function],
      "formatNumber": [Function],
      "formatPlural": [Function],
      "formatRelative": [Function],
      "formatTime": [Function],
      "formats": Object {},
      "formatters": Object {
        "getDateTimeFormat": [Function],
        "getMessageFormat": [Function],
        "getNumberFormat": [Function],
        "getPluralFormat": [Function],
        "getRelativeFormat": [Function],
      },
      "locale": "en",
      "messages": Object {},
      "now": [Function],
      "textComponent": "span",
      "timeZone": null,
    }
  }
  isExpanded={false}
  isLoading={true}
  onNavigate={[MockFunction]}
  onOpen={[Function]}
  value=""
>
  <ProductSearchInput
    actionSubjectId="jiraIssuesSearchLink"
    advancedSearchURL="/secure/QuickSearch.jspa?searchString="
    collapsedPlaceholder="Search"
    debounceTime={0}
    expandedPlaceholder="Search Jira"
    forwardRef={null}
    isExpanded={false}
    isLoading={true}
    onNavigate={[MockFunction]}
    onOpen={[Function]}
    value=""
  >
    <div
      actionSubjectId="jiraIssuesSearchLink"
      advancedSearchURL="/secure/QuickSearch.jspa?searchString="
      collapsedPlaceholder="Search"
      debounceTime={0}
      expandedPlaceholder="Search Jira"
      forwardRef={null}
      isExpanded={false}
      isLoading={true}
      onNavigate={[MockFunction]}
      onOpen={[Function]}
      value=""
    />
  </ProductSearchInput>
</SearchInputBase>
`;

exports[`<SearchInput /> matches snapshot when expanded 1`] = `
<SearchInputBase
  debounceTime={0}
  forwardRef={null}
  intl={
    Object {
      "defaultFormats": Object {},
      "defaultLocale": "en",
      "formatDate": [Function],
      "formatHTMLMessage": [Function],
      "formatMessage": [Function],
      "formatNumber": [Function],
      "formatPlural": [Function],
      "formatRelative": [Function],
      "formatTime": [Function],
      "formats": Object {},
      "formatters": Object {
        "getDateTimeFormat": [Function],
        "getMessageFormat": [Function],
        "getNumberFormat": [Function],
        "getPluralFormat": [Function],
        "getRelativeFormat": [Function],
      },
      "locale": "en",
      "messages": Object {},
      "now": [Function],
      "textComponent": "span",
      "timeZone": null,
    }
  }
  isExpanded={true}
  isLoading={true}
  onNavigate={[MockFunction]}
  onOpen={[Function]}
  value=""
>
  <ProductSearchInput
    actionSubjectId="jiraIssuesSearchLink"
    advancedSearchURL="/secure/QuickSearch.jspa?searchString="
    collapsedPlaceholder="Search"
    debounceTime={0}
    expandedPlaceholder="Search Jira"
    forwardRef={null}
    isExpanded={true}
    isLoading={true}
    onNavigate={[MockFunction]}
    onOpen={[Function]}
    value=""
  >
    <div
      actionSubjectId="jiraIssuesSearchLink"
      advancedSearchURL="/secure/QuickSearch.jspa?searchString="
      collapsedPlaceholder="Search"
      debounceTime={0}
      expandedPlaceholder="Search Jira"
      forwardRef={null}
      isExpanded={true}
      isLoading={true}
      onNavigate={[MockFunction]}
      onOpen={[Function]}
      value=""
    />
  </ProductSearchInput>
</SearchInputBase>
`;
